src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -68,54 +64,34 @@
     private final Path dir;
 
     /**
-     * IO executor service.
-     */
-    private final ExecutorService exec;
-
-    /**
-     * Ctor.
-     * @param path The path to the dir
-     * @param nothing Just for compatibility
-     * @deprecated Use {@link FileStorage#FileStorage(Path)} ctor instead.
-     */
-    @Deprecated
-    @SuppressWarnings("PMD.UnusedFormalParameter")
-    public FileStorage(final Path path, final Object nothing) {
-        this(path);
-    }
-
-    /**
-     * Ctor.
-     * @param path File path
+     * The Vert.x file system.
      */
-    public FileStorage(final Path path) {
-        this(path, ThreadPool.EXEC);
-    }
+    private final FileSystem fls;
 
     /**
      * Ctor.
+     *
      * @param path The path to the dir
-     * @param exec IO Executor service
+     * @param fls The file system
      */
-    public FileStorage(final Path path, final ExecutorService exec) {
+    public FileStorage(final Path path, final FileSystem fls) {
         this.dir = path;
-        this.exec = exec;
+        this.fls = fls;
     }
 
     @Override
     public CompletableFuture<Boolean> exists(final Key key) {
-        return CompletableFuture.supplyAsync(
+        return Single.fromCallable(
```

Comment:
@Sammers21 what is the reason for this change?

Reply:
@olegmoz, the `CompletableFuture.supplyAsync` call does not allow us to use vert.x as something to delegate work to

Reply:
@Sammers21 I am sorry but why do we need to give this kind of tasks to Vert.x? And how `Single.fromCallable` is related to Vert.x?

Reply:
@olegmoz, you can delegate tasks to vert.x blocking executor